pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -289,84 +304,93 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
-			r.status.SetDegraded(operatorv1.ResourcePatchError, fmt.Sprintf("Failed to set defaults for LogCollector fields: [%s]",
-				strings.Join(modifiedFields, ", "),
-			), err, reqLogger)
+	// Fetch the Installation instance. We need this for a few reasons.
+	// - We need to make sure it has successfully completed installation.
+	// - We need to get the registry information from its spec.
+	variant, installation, err := utils.GetInstallation(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Installation not found", err, reqLogger)
 			return reconcile.Result{}, err
 		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if variant != operatorv1.TigeraSecureEnterprise && (instance.Spec.AdditionalStores != nil || instance.Spec.CollectProcessPath != nil) {
+		r.status.SetDegraded(operatorv1.InvalidConfigurationError, "Additional log sources can be configured only in calico enterprise.", nil, reqLogger)
+		return reconcile.Result{}, fmt.Errorf("Additional log sources can be configured only in calico enterprise.")
 	}
 
 	if !utils.IsAPIServerReady(r.client, reqLogger) {
 		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Tigera API server to be ready", nil, reqLogger)
 		return reconcile.Result{}, nil
 	}
 
-	// Validate that the tier watch is ready before querying the tier to ensure we utilize the cache.
-	if !r.tierWatchReady.IsReady() {
-		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Tier watch to be established", nil, reqLogger)
-		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
-	}
-
-	// Ensure the allow-tigera tier exists, before rendering any network policies within it.
-	if err := r.client.Get(ctx, client.ObjectKey{Name: networkpolicy.TigeraComponentTierName}, &v3.Tier{}); err != nil {
-		if errors.IsNotFound(err) {
-			r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for allow-tigera tier to be created", err, reqLogger)
+	if variant == operatorv1.TigeraSecureEnterprise {
+		// Validate that the tier watch is ready before querying the tier to ensure we utilize the cache.
+		if !r.tierWatchReady.IsReady() {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Tier watch to be established", nil, reqLogger)
 			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
-		} else {
-			log.Error(err, "Error querying allow-tigera tier")
-			r.status.SetDegraded(operatorv1.ResourceNotReady, "Error querying allow-tigera tier", err, reqLogger)
-			return reconcile.Result{}, err
 		}
-	}
 
-	if !r.licenseAPIReady.IsReady() {
-		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for LicenseKeyAPI to be ready", nil, reqLogger)
-		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
-	}
+		// Default fields on the LogCollector instance if needed.
```

Comment:
For logCollector in OSS we are not going to set any of the fields in spec.Hence defaulting only for EE

Reply:
Even so, we need to call the defaulting function. 

We should update the defaulting logic so that for OSS, we either don't default or set different defaults, but we shouldn't bypass the defaulting function altogether.

Conceptually, defaulting isn't an enterprise-only thing.